Screens/PerformanceScreen.js

```diff
@@ -109,11 +130,21 @@ const styles = StyleSheet.create({
     },
 });
 
-const mapStateToProps = state => {
+const mapStateToProps = (state) => {
+    const roles = _.groupBy(state.performanceReducer.performance.teamMember, teamMember => teamMember.roleKey);
```

Comment:
rename teamMember to teamMembers
Also put grouping inside selector

You can return from getCreativeTeamMembers following object
{
  Authors,
  Producers,
  Composers,
  Painters,
}

and then assign it to props using spread operator ...